From 996367c9bb3d8dae69654e88adeecb163ab0b282 Mon Sep 17 00:00:00 2001
From: Scott Todd <scott.todd0@gmail.com>
Date: Thu, 5 Jun 2025 17:48:02 -0700
Subject: [PATCH 3/3] Unify HIPBLASLT_TENSILE_LIBRARY_DIR across Windows and
 Linux.

Splitting the structure leads to further complexity in all downstream projects and tooling.
---
 library/CMakeLists.txt    | 6 +-----
 next-cmake/CMakeLists.txt | 6 +-----
 2 files changed, 2 insertions(+), 10 deletions(-)

diff --git a/library/CMakeLists.txt b/library/CMakeLists.txt
index ccda26c5..5195f7d3 100644
--- a/library/CMakeLists.txt
+++ b/library/CMakeLists.txt
@@ -238,11 +238,7 @@ install(
 )
 
 if ( NOT BUILD_CUDA )
-    if (WIN32)
-      set( HIPBLASLT_TENSILE_LIBRARY_DIR "\${CPACK_PACKAGING_INSTALL_PREFIX}hipblaslt/bin" CACHE PATH "path to tensile library" )
-    else()
-      set( HIPBLASLT_TENSILE_LIBRARY_DIR "\${CPACK_PACKAGING_INSTALL_PREFIX}${CMAKE_INSTALL_LIBDIR}/hipblaslt" CACHE PATH "path to tensile library" )
-    endif()
+    set( HIPBLASLT_TENSILE_LIBRARY_DIR "\${CPACK_PACKAGING_INSTALL_PREFIX}${CMAKE_INSTALL_LIBDIR}/hipblaslt" CACHE PATH "path to tensile library" )
 # For ASAN package, Tensile library files are not required
     if( NOT ENABLE_ASAN_PACKAGING )
     if( BUILD_SHARED_LIBS )
diff --git a/next-cmake/CMakeLists.txt b/next-cmake/CMakeLists.txt
index 3ba8eabc..0417749b 100644
--- a/next-cmake/CMakeLists.txt
+++ b/next-cmake/CMakeLists.txt
@@ -328,11 +328,7 @@ if(HIPBLASLT_ENABLE_HOST)
 endif()
 
 if(HIPBLASLT_ENABLE_DEVICE)
-    if(WIN32)
-        set(HIPBLASLT_TENSILE_LIBRARY_DIR "\${CPACK_PACKAGING_INSTALL_PREFIX}hipblaslt/bin" CACHE PATH "path to tensile library")
-    else()
-        set(HIPBLASLT_TENSILE_LIBRARY_DIR "\${CPACK_PACKAGING_INSTALL_PREFIX}${CMAKE_INSTALL_LIBDIR}/hipblaslt" CACHE PATH "path to tensile library")
-    endif()
+    set(HIPBLASLT_TENSILE_LIBRARY_DIR "\${CPACK_PACKAGING_INSTALL_PREFIX}${CMAKE_INSTALL_LIBDIR}/hipblaslt" CACHE PATH "path to tensile library")
     rocm_install(
         DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/Tensile/library
         DESTINATION ${HIPBLASLT_TENSILE_LIBRARY_DIR}
-- 
2.45.1.windows.1

